virttest/test_setup/__init__.py

```diff
@@ -818,7 +818,11 @@ def __init__(self, params=None):
             u_port = params.get("guest_port_unattended_install", "13323")
             if u_port not in ports:
                 ports.append(u_port)
-            self.iptables_rules = self._assemble_iptables_rules(ports)
+            self.firewall_tool = self._check_firewall_tool()
+            if self.firewall_tool == "iptables":
+                self.iptables_rules = self._assemble_iptables_rules(ports)
+            else:
+                self.iptables_rules = self._assemble_nftables_rules(ports)
```

Comment:
Hi @quanwenli 
What's the policy for selecting the rules?
In other words, How can I determine the default rules when both `iptables` and `nftables` exist on the system?

From my researching, choosing between `iptables` and `nftables` depends on your specific requirements, the complexity of your network configurations, and your performance needs. For new projects or when upgrading existing systems, `nftables` is generally the preferred choice.

These are just my personal thoughts. I would like to hear your perspective. Thanks.

Reply:
Iptables was the initial tool used, and then nftables was introduced. For a while, both existed at the same time. In the latest systems, iptables has been retired. I create this pull request since it block the test in latest system. I think we can keep using iptables even if both iptables and nftables are available, as it would enhance the stability of the code.

For perf testing, I don’t care which tool is used since it will be flushed during the test case execution.

Reply:
For individual test cases, choosing tools may depend on the test developer. Some developers may not care about the tool used, while others may have strong preferences. Therefore, we should not set `iptables` as the default mandatory tool. Instead, it is better to base this decision on specific conditions.

Hi, @leidwang 
What do you think about choosing the tool to add the rules?
CC @fbq815 @zhenyzha 
Thanks.

Reply:
Hi @YongxueHong I agree with @quanwenli point, I don't care which tools are used in testing, just it can support to add rules is ok for me. The iptables is deprecate from rhel 9,however the rhel 9 still an find `iptable_filter.ko.xz`, I think we should re-write choosing tool's rule.

Reply:
Hi @YongxueHong @quanwenli Let me correct my points, from my side if using iptables on rhel 9 it will hit automation bug since host dmesg will print like this `Warning: Unmaintained driver is detected: nft_compat` error messges. Therefore, based on this, which tools will be used is important to me.  So @quanwenli , could you please help re-write this function `_check_firewall_tool(self):` ? I think based on the host version to choosing tools is better than find a iptables's file.

Reply:
Firstly, according to the article at https://access.redhat.com/solutions/6739041, iptables is still supported in RHEL9, and the warning message is considered acceptable.

Secondly, as detailed in the 'Migrating from iptables to nftables' link, it's clear that:
  - firewalld should be used for simple firewall configurations.
  -nftables should be employed for more complex and performance-critical setups, such as those required for an entire network.

Finally, at begining design , I considered  using the host version to determine support for nftables. However, it's unclear from which version nftables should be supported. so, I think from module perspective, choosing modules  is more acceptable and accurate

Reply:
@YongxueHong since iptables is deprecated on 10 I agree to use nftables on 10 and later